src/compiler/crystal/semantic/lib.cr

```diff
@@ -57,11 +57,11 @@ class Crystal::Call
     named_args.each do |named_arg|
       found_index = external.args.index { |arg| arg.name == named_arg.name }
       unless found_index
-        named_arg.raise "no argument named '#{named_arg.name}'"
+        named_arg.raise "no parameter named '#{named_arg.name}'"
       end
 
       if covered[found_index]
-        named_arg.raise "argument '#{named_arg.name}' already specified"
+        named_arg.raise "argument for parameter '#{named_arg.name}' already specified"
```

Comment:
Is "for parameter" really helpful to improve this message?

Reply:
Parameters always have a name but this isn't the case for arguments (in some other error messages they are referenced by numbers like `#1` and so on).

Though using "parameter ... already specified" here is also fine to me.

Reply:
But this is about a call, so it's a duplicate argument, not a duplicate parameter.

Reply:
There cannot be duplicate arguments. Positional arguments are always identifiable by unique indices and duplicate named arguments are detected by the parser.

Reply:
Technically this is correct, yes. But from a user's perspective I would prefer not to be that specific here.
`Argument 'foo' already specified` is a completely sufficient error message.

Reply:
Technical correctness is the whole point of this issue. I don't see why this particular error message would be one that users prefer to be less specific than the other messages involving arguments.

Reply:
Why would shortening `argument` to `argument for parameter` be less correct? It's just more concise. Would the shortened form cause any ambiguity or misunderstanding of the error message's meaning?